packages/core/src/telemetry/config.ts

```
@@ -34,10 +34,158 @@ export function parseTelemetryTargetValue(
+  }
+  // RFC 7230: Valid header name characters
+  return /^[A-Za-z0-9!#$%&'*+\-.^_`|~]+$/.test(name);
+};
+
+const isValidHeaderValue = (value: string): boolean => {
+  if (value.length > MAX_HEADER_VALUE_LENGTH) {
+    return false;
+  }
+  // RFC 7230: No control characters except tab
+  // eslint-disable-next-line no-control-regex
+  return !/[\x00-\x08\x0A-\x1F\x7F]/.test(value);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)